### ISSUE TITLE
nn.L1Loss works incorrectly with CPU

### ISSUE THREAD

**Issue by CastellanLiu** [pytorch/pytorch#27692]
## 🐛 Bug

When a tensor is large, the L1 loss is not equal to `torch.abs(x - y).mean()` in CPU mode, but it works fine in GPU mode.

I find this bug in PyTorch 1.2.0, and have verified that this problem exists in PyTorch 0.4.0 (on centos).

## To Reproduce

Just create a large tensor, and test with the following code. I can reproduce it with `Intel i7-8750H` and `Intel E3-1230 v5`.

```python
import torch
# NOTE that in this situation, when N is less than 10, the results are usually
# equal or similar, and the difference becomes larger when N is increased
a = torch.rand(12, 3, 1024, 1024)
print(torch.nn.L1Loss()(a, torch.zeros_like(a)))
print(a.abs().mean())
```

## Environment

PyTorch version: 1.2.0
Is debug build: No
CUDA used to build PyTorch: 10.0

OS: Microsoft Windows 10 Pro
GCC version: Could not collect
CMake version: Could not collect

Python version: 3.6
Is CUDA available: Yes
CUDA runtime version: 10.0.130
GPU models and configuration: Could not collect
Nvidia driver version: Could not collect
cuDNN version: C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v10.0\bin\cudnn64_7.dll

Versions of relevant libraries:
[pip3] numpy==1.17.2
[pip3] numpydoc==0.7.0
[pip3] torch==1.2.0
[pip3] torchfile==0.1.0
[pip3] torchvision==0.4.0
[conda] blas                      1.0                         mkl    defaults
[conda] mkl                       2018.0.1             h2108138_4    defaults
[conda] mkl-service               1.1.2            py36h57e144c_4    defaults
[conda] pytorch                   1.2.0           py3.6_cuda100_cudnn7_1    defaults
[conda] torchfile                 0.1.0                    pypi_0    pypi
[conda] torchvision               0.4.0                py36_cu100    defaults

cc @ezyang @gchanan @zou3519 @jerryzh168 @VitalyFedyunin

**Comment by CastellanLiu** [comment]
This problem exists on Windows 10, Ubuntu 18.04, Ubuntu 14.04 and Centos 6.8, so I guess it's unrelated to system platform.

**Comment by XiaobingSuper** [comment]
@CastellanLiu, there a PR(https://github.com/pytorch/pytorch/pull/26795) can solve this problem, which is under reviewing, you can try it first.  @VitalyFedyunin, @zou3519, please help see this problem. Thanks!

**Comment by CastellanLiu** [comment]
Thanks for your reply, I find this bug by chance, and I'm using the GPU mode at present.
I will try this PR when I have some spare time.

**Comment by cpuhrsch** [comment]
I'm flagging this for triage review if the related PR hasn't been reviewed or landed by the time we review this issue again.

**Comment by ezyang** [comment]
Reproduces on master:

```
$ python test.py 
tensor(0.4444)
tensor(0.5000)
```

I'm checking if the l1 loss port fixes it.

**Comment by ezyang** [comment]
No, the l1 loss port does not fix it.

```
(/scratch/ezyang/pytorch-tmp-env) ezyang@devfair024:/scratch/ezyang/pytorch-tmp$ python test.py
tensor(0.4444)
tensor(0.5001)
(/scratch/ezyang/pytorch-tmp-env) ezyang@devfair024:/scratch/ezyang/pytorch-tmp$ git rev-parse HEAD
a6348e02ce4673296074bd48d9e8f0597e85d950
```

**Comment by XiaobingSuper** [comment]
@ezyang , I double check the PR(#26795)'s result, I can make sure it can solve this problem, please see the follwing log:
```
pytorch-master) [xiaobinz@mlt-clx063 pytorch-xiaobing]$ conda list|grep torch
# packages in environment at /home/xiaobinz/anaconda3/envs/pytorch-master:
torch                     1.4.0a0+22dbefc          pypi_0    pypi
(pytorch-master) [xiaobinz@mlt-clx063 pytorch-xiaobing]$ cd test/
(pytorch-master) [xiaobinz@mlt-clx063 test]$ python
Python 3.6.9 |Anaconda, Inc.| (default, Jul 30 2019, 19:07:31)
[GCC 7.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import torch
>>> a = torch.rand(12, 3, 1024, 1024)
>>> print(torch.nn.L1Loss()(a, torch.zeros_like(a)))
tensor(0.5000)
>>> print(a.abs().mean())
tensor(0.5000)
```

**Comment by ezyang** [comment]
Interesting. Are you saying my build is out of date? In any case, if you add a test for this, I'll trust you on it :)

**Comment by XiaobingSuper** [comment]
@ezyang, #26795 has been merged which can solve this problem, we can close this issue now. Thanks!
